WARNING:root:You are using an untested version of z3. 4.5.0 is the officially tested version
WARNING:root:You are using solc version 0.4.25, The latest supported version is 0.4.17
INFO:root:Contract smart_contract_source_code/2301.sol:BlockHashRNG:
INFO:symExec:Running, please wait...
INFO:symExec:	============ Results ===========
INFO:symExec:	  EVM code coverage: 	 99.9%
INFO:symExec:	  Callstack bug: 	 False
INFO:symExec:	  Money concurrency bug: True
Flow 1:
smart_contract_source_code/2301.sol:BlockHashRNG:578:13
msg.sender.send(rewardToSend)
^
Flow 2:
smart_contract_source_code/2301.sol:BlockHashRNG:578:13
msg.sender.send(rewardToSend)
^
INFO:symExec:	  Time dependency bug: 	 False
INFO:symExec:	  Reentrancy bug: 	 False
INFO:root:Contract smart_contract_source_code/2301.sol:BlockHashRNGFallback:
INFO:symExec:Running, please wait...
INFO:symExec:	============ Results ===========
INFO:symExec:	  EVM code coverage: 	 99.9%
INFO:symExec:	  Callstack bug: 	 False
INFO:symExec:	  Money concurrency bug: True
Flow 1:
smart_contract_source_code/2301.sol:BlockHashRNGFallback:599:13
msg.sender.send(rewardToSend)
^
Flow 2:
smart_contract_source_code/2301.sol:BlockHashRNGFallback:599:13
msg.sender.send(rewardToSend)
^
INFO:symExec:	  Time dependency bug: 	 False
INFO:symExec:	  Reentrancy bug: 	 False
INFO:root:Contract smart_contract_source_code/2301.sol:Controlled:
INFO:symExec:Running, please wait...
INFO:symExec:	============ Results ===========
INFO:symExec:	  EVM code coverage: 	 99.4%
INFO:symExec:	  Callstack bug: 	 False
INFO:symExec:	  Money concurrency bug: False
INFO:symExec:	  Time dependency bug: 	 False
INFO:symExec:	  Reentrancy bug: 	 False
INFO:root:Contract smart_contract_source_code/2301.sol:Kleros:
INFO:symExec:Running, please wait...
INFO:symExec:	============ Results ===========
Traceback (most recent call last):
  File "/home/jion1/Downloads/oyente-0.2.7/oyente/symExec.py", line 2359, in main
    build_cfg_and_analyze()
  File "/home/jion1/Downloads/oyente-0.2.7/oyente/symExec.py", line 211, in build_cfg_and_analyze
    collect_vertices(tokens)
  File "/home/jion1/Downloads/oyente-0.2.7/oyente/symExec.py", line 327, in collect_vertices
    idx = mapping_non_push_instruction(current_line_content, current_ins_address, idx, positions, length) if source_map else None
  File "/home/jion1/Downloads/oyente-0.2.7/oyente/symExec.py", line 267, in mapping_non_push_instruction
    raise Exception("Source map error")
Exception: Source map error
Traceback (most recent call last):
  File "oyente.py", line 256, in <module>
    main()
  File "oyente.py", line 242, in main
    analyze(processed_evm_file, disasm_file, SourceMap(cname, args.source))
  File "oyente.py", line 109, in analyze
    symExec.main(disasm_file, args.source, source_map)
  File "/home/jion1/Downloads/oyente-0.2.7/oyente/symExec.py", line 2366, in main
    raise e
Exception: Source map error
INFO:symExec:	====== Analysis Completed ======
INFO:symExec:	====== Analysis Completed ======
INFO:symExec:	====== Analysis Completed ======
